pkgs/development/libraries/qt-6/modules/qtwebengine.nix

```diff
@@ -0,0 +1,234 @@
+{ qtModule
+, qtdeclarative
+, qtwebchannel
+, qtpositioning
+, qtwebsockets
+, bison
+, coreutils
+, flex
+, git
+, gperf
+, ninja
+, pkg-config
+, python3
+, which
+, nodejs
+, qtbase
+, perl
+, xorg
+, libXcursor
+, libXScrnSaver
+, libXrandr
+, libXtst
+, libxshmfence
+, libXi
+, fontconfig
+, freetype
+, harfbuzz
+, icu
+, dbus
+, libdrm
+, zlib
+, minizip
+, libjpeg
+, libpng
+, libtiff
+, libwebp
+, libopus
+, jsoncpp
+, protobuf
+, libvpx
+, srtp
+, snappy
+, nss
+, libevent
+, openssl
+, alsa-lib
+, pulseaudio
+, libcap
+, pciutils
+, systemd
+, pipewire
+, gn
+, cups
+, openbsm
+, runCommand
+, writeScriptBin
+, ffmpeg
+, lib
+, stdenv
+, fetchpatch
+, glib
+, libxml2
+, libxslt
+, lcms2
+, re2
+, kerberos
+, enableProprietaryCodecs ? true
+}:
+
+qtModule rec {
+  pname = "qtwebengine";
+  qtInputs = [ qtdeclarative qtwebchannel qtwebsockets qtpositioning ];
+  nativeBuildInputs = [
+    bison
+    coreutils
+    flex
+    git
+    gperf
+    ninja
+    pkg-config
+    (python3.withPackages (ps: with ps; [ html5lib ]))
+    which
+    gn
+    nodejs
+  ];
+  doCheck = true;
+  outputs = [ "out" "dev" ];
+
+  dontUseGnConfigure = true;
+
+  # ninja builds some components with -Wno-format,
+  # which cannot be set at the same time as -Wformat-security
+  hardeningDisable = [ "format" ];
+
+  patches = [
+    (fetchpatch {
```

Comment:
What does this patch do?

Reply:
Otherwise it fails to build with system icu.

Reply:
Right. Can you add a comment explaining that? That's what we normally do for patches like this.